nautobot/dcim/management/commands/migrate_location_contacts.py

```diff
@@ -0,0 +1,205 @@
+from textwrap import indent
+
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from nautobot.dcim.models import Location
+from nautobot.extras.filters import ContactFilterSet, TeamFilterSet
+from nautobot.extras.models import Contact, ContactAssociation, Role, Status, Team
+
+
+class Command(BaseCommand):
+    help = "Migrate Location contact fields to Contact and Team objects."
+
+    def handle(self, *args, **kwargs):
+        try:
+            with transaction.atomic():
+                try:
+                    self.migrate_location_contacts()
+                except KeyboardInterrupt:
+                    while True:
+                        rollback = input("\nRoll back changes? [y/n]: ").strip().lower()
+                        if rollback == "y":
+                            raise
+                        elif rollback == "n":
+                            break
+        except KeyboardInterrupt:
+            self.stdout.write(self.style.ERROR("\nOperation cancelled, all changes rolled back."))
+        except:
+            self.stdout.write(self.style.ERROR("\nOperation failed, all changes rolled back."))
+            raise
+
+    def migrate_location_contacts(self):
+        """Iterate through Locations with contact information and try to match to existing Contact or Team."""
+        locations_with_contact_data = Location.objects.exclude(
+            contact_name__isnull=True,
+            contact_phone__isnull=True,
+            contact_email__isnull=True,
+        ).filter(associated_contacts__isnull=True)
+
+        for location in locations_with_contact_data:
+            self.stdout.write(f"Finding existing Contacts or Teams for location {location.display}...")
+            selected_contact = None
+            similar_contacts = list(ContactFilterSet(data={"similar_to_location_data": [location]}).qs)
+            similar_teams = list(TeamFilterSet(data={"similar_to_location_data": [location]}).qs)
+            similar_contacts_and_teams = similar_contacts + similar_teams
+
+            if not similar_contacts_and_teams:
+                self.stdout.write(
+                    self.style.WARNING(f"No similar Contacts or Teams found for location {location.display}.")
+                )
+                self.stdout.write(self.style.WARNING("c") + ": Create a new Contact")
+                self.stdout.write(self.style.WARNING("t") + ": Create a new Team")
+                self.stdout.write(self.style.WARNING("s") + ": Skip this location")
```

Comment:
Maybe support (but don't need to print/display) "q" as an alternative to ctrl-C?